src/pages/index.js

```diff
@@ -73,29 +157,79 @@ const newCardPopup = new PopupWithForm(
 newCardPopup.setEventListeners();
 
 function handleAddNewCardSubmit(data) {
-  const card = createCard({ name: data.name, link: data.url });
-  cardSection.addItem(card);
+  // const card = createCard({ name: data.name, link: data.url });
+  newCardPopup.submitText("Saving...");
+  api
+    .createCard(data)
+    .then((data) => createCard(data))
+    .then((data) => cardSection.addItem(data))
+    .then(() => newCardPopup.close())
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => newCardPopup.submitText("Save"));
+
   // cardFormValidator.resetValidation();
-  newCardPopup.close();
 }
-function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", () =>
-    handleImageClick(cardData)
-  );
-  return card.getView();
+
+const newAvatarPopup = new PopupWithForm(
+  "#avatar-modal",
+  handleNewAvatarSubmit
+);
+newAvatarPopup.setEventListeners();
+
+function handleNewAvatarSubmit(data) {
+  newAvatarPopup.submitText("Saving...");
+  api
+    .editAvatar(data)
+    .then(() => {
+      userInfo.setUserInfo({
+        name: profileName.textContent,
+        job: profileDescription.textContent,
```

Comment:
`textContent` shouldn't be used anymore in `index.js`. You get the whole data in the **response** from the server


```js

    .then((userData) => {
      userInfo.setUserInfo({
        name: userData.name,
        job: userData.about,
        avatar: userData.avatar,
      });
```